pkg/controller/atlasdeployment/atlasdeployment_controller_test.go

```diff
@@ -1090,3 +1093,334 @@ func TestFindDeploymentsForBackupSchedule(t *testing.T) {
+			atlasDeployment: &akov2.AtlasDeployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "cluster0",
+					Namespace: "default",
+				},
+				Spec: akov2.AtlasDeploymentSpec{
+					ExternalProjectRef: &akov2.ExternalProjectReference{
+						ID: "project-id",
+					},
+					LocalCredentialHolder: api.LocalCredentialHolder{
+						ConnectionSecret: &api.LocalObjectReference{
+							Name: "project-creds",
+						},
+					},
+				},
+			},
+			atlasProvider: &atlasmock.TestProvider{
+				IsCloudGovFunc: func() bool {
+					return false
+				},
+				SdkClientFunc: func(secretRef *client.ObjectKey, log *zap.SugaredLogger) (*admin.APIClient, string, error) {
+					projectAPI := mockadmin.NewProjectsApi(t)
+					projectAPI.EXPECT().GetProject(context.Background(), "project-id").
+						Return(admin.GetProjectApiRequest{ApiService: projectAPI})
+					projectAPI.EXPECT().GetProjectExecute(mock.AnythingOfType("admin.GetProjectApiRequest")).
+						Return(nil, nil, errors.New("failed to get project"))
+
+					return &admin.APIClient{ProjectsApi: projectAPI}, "", nil
+				},
+			},
+			expectedErr: errors.New("failed to get project"),
+		},
+		"get project": {
+			atlasDeployment: &akov2.AtlasDeployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "user1",
+					Namespace: "default",
+				},
+				Spec: akov2.AtlasDeploymentSpec{
+					ExternalProjectRef: &akov2.ExternalProjectReference{
+						ID: "project-id",
+					},
+					LocalCredentialHolder: api.LocalCredentialHolder{
+						ConnectionSecret: &api.LocalObjectReference{
+							Name: "project-creds",
+						},
+					},
+				},
+			},
+			deploymentSecret: &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "user-pass",
+					Namespace: "default",
+					Labels: map[string]string{
+						"atlas.mongodb.com/type": "credentials",
+					},
+				},
+				Data: map[string][]byte{
+					"password": []byte("Passw0rd!"),
+				},
+			},
+			atlasProvider: &atlasmock.TestProvider{
+				IsCloudGovFunc: func() bool {
+					return false
+				},
+				SdkClientFunc: func(secretRef *client.ObjectKey, log *zap.SugaredLogger) (*admin.APIClient, string, error) {
+					projectAPI := mockadmin.NewProjectsApi(t)
+					projectAPI.EXPECT().GetProject(context.Background(), "project-id").
+						Return(admin.GetProjectApiRequest{ApiService: projectAPI})
+					projectAPI.EXPECT().GetProjectExecute(mock.AnythingOfType("admin.GetProjectApiRequest")).
+						Return(&admin.Group{Id: pointer.MakePtr("project-id")}, nil, nil)
+
+					return &admin.APIClient{ProjectsApi: projectAPI}, "", nil
+				},
+				ClientFunc: func(secretRef *client.ObjectKey, log *zap.SugaredLogger) (*mongodbatlas.Client, string, error) {
+					return &mongodbatlas.Client{}, "", nil
+				},
+			},
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			testScheme := runtime.NewScheme()
+			assert.NoError(t, akov2.AddToScheme(testScheme))
+			assert.NoError(t, corev1.AddToScheme(testScheme))
```

Comment:
These could happen once and outside of the loop and sub test right?

Reply:
Yes, but it would be a minor optimization. We follow this pattern in every test we have.